src/HaskellCI.hs

```diff
@@ -761,18 +743,24 @@ genTravisFromConfigs argv opts isCabalProject config prj@Project { prjPackages =
         tellStrLns
             [ comment "Constraint sets"
             , sh "rm -rf cabal.project.local"
-            , ""
             ]
         forM_ constraintSets $ \cs -> do
             let name = csName cs
-            let constraintFlags = concatMap (\x ->  " --constraint='" ++ x ++ "'") (csConstraints cs)
-            let cmd | csRunTests cs = "${CABAL} new-test  -w ${HC} --enable-tests  --enable-benchmarks"
-                    | otherwise     = "${CABAL} new-build -w ${HC} --disable-tests --disable-benchmarks"
             tellStrLns [ comment $ "Constraint set " ++ name ]
+
+            let shForCs = shForJob versions' (csGhcVersions cs)
+            let constraintFlags = concatMap (\x ->  " --constraint='" ++ x ++ "'") (csConstraints cs)
+
             foldedTellStrLns' FoldConstraintSets name ("Constraint set " ++ name) folds $ tellStrLns
-                [ shForJob versions' (csGhcVersions cs) $
-                  cmd ++ " " ++ constraintFlags ++ " all"
-                , ""
+                [ shForCs $ "CSTEST=" ++ if csTests cs then "--enable-tests" else "--disable-tests"
+                , shForCs $ "CSBENCH=" ++ if csBenchmarks cs then "--enable-benchmarks" else "--disable-benchmarks"
+                , shForCs $ "${CABAL} v2-build -w ${HC} ${CSTEST} ${CSBENCH} all " ++ constraintFlags
```

Comment:
This is a really minor thing, but is there any reason to use `v2-` here over `new-`, like the rest of `haskell-ci` uses?

Reply:
I'll slowly convert everything to `v2-` at some point.